akka-actor/src/main/scala/akka/pattern/StatusReply.scala

```diff
@@ -118,7 +118,7 @@ object StatusReply {
      */
     def apply[T](value: T): StatusReply[T] = new StatusReply(ScalaSuccess(value))
     def unapply(status: StatusReply[Any]): Option[Any] =
-      if (status.isSuccess) Some(status.getValue)
+      if (status != null && status.isSuccess) Some(status.getValue)
```

Comment:
Just the change in the unnapplies is enough, we don't need to _also_ do the change when using the custom extractor in the pattern match?